server/src/main/java/org/apache/seata/server/cluster/raft/snapshot/metadata/ConfigLeaderMetadataSnapshotFile.java

```diff
@@ -0,0 +1,88 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.seata.server.cluster.raft.snapshot.metadata;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+
+import com.alipay.sofa.jraft.Status;
+import com.alipay.sofa.jraft.error.RaftError;
+import com.alipay.sofa.jraft.storage.snapshot.SnapshotReader;
+import com.alipay.sofa.jraft.storage.snapshot.SnapshotWriter;
+import org.apache.seata.server.cluster.raft.RaftConfigServerManager;
+import org.apache.seata.server.cluster.raft.snapshot.RaftSnapshot;
+import org.apache.seata.server.cluster.raft.snapshot.StoreSnapshotFile;
+import org.apache.seata.server.cluster.raft.sync.msg.dto.RaftClusterMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigLeaderMetadataSnapshotFile implements Serializable, StoreSnapshotFile {
+    private static final long serialVersionUID = 43235664615355354L;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigLeaderMetadataSnapshotFile.class);
+
+    private final String group;
+
+    private final String fileName = "leader_metadata";
+
+    public ConfigLeaderMetadataSnapshotFile(String group) {
+        this.group = group;
+    }
+
+
+    @Override
+    public Status save(SnapshotWriter writer) {
+        RaftSnapshot raftSnapshot = new RaftSnapshot();
+        RaftClusterMetadata raftClusterMetadata =
+                RaftConfigServerManager.getRaftServer().getRaftStateMachine().getRaftLeaderMetadata();
+        raftSnapshot.setBody(raftClusterMetadata);
+        raftSnapshot.setType(RaftSnapshot.SnapshotType.leader_metadata);
+        String path = new StringBuilder(writer.getPath()).append(File.separator).append(fileName).toString();
+        try {
+            if (save(raftSnapshot, path)) {
+                if (writer.addFile(fileName)) {
+                    return Status.OK();
+                } else {
+                    return new Status(RaftError.EIO, "Fail to add file to writer");
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.error("Fail to save groupId: {} snapshot {}", group, path, e);
+        }
+        return new Status(RaftError.EIO, "Fail to save groupId: " + group + " snapshot %s", path);
+    }
+
+    @Override
+    public boolean load(SnapshotReader reader) {
+        if (reader.getFileMeta(fileName) == null) {
+            LOGGER.error("Fail to find data file in {}", reader.getPath());
+            return false;
+        }
+        String path = new StringBuilder(reader.getPath()).append(File.separator).append(fileName).toString();
+        try {
+            RaftClusterMetadata raftClusterMetadata = (RaftClusterMetadata)load(path);
+            RaftConfigServerManager.getRaftServer().getRaftStateMachine()
+                    .setRaftLeaderMetadata(raftClusterMetadata);
+            return true;
+        } catch (final Exception e) {
```

Comment:
Is 'final' here necessary?

Reply:
just consistent with other snapshot class code, or uniformly remove the 'final' in all snapshot class